### PR TITLE
[local_auth] Overhaul README, and fix `error_codes.dart` visibility

### PR DESCRIPTION
**Pull request by stuartmorgan**
Improves the README on a number of fronts:
- Converts the many code snippets to use the new `code-excerpt` system,
  fixing a number of obsolete APIs that would not even compile against 2.0,
  and generally improving their clarity and consistency.
- Updates the section about Android activity updates to use v2, not v1, examples.
- Made most discussions of biometrics more general, instead of referring
  to specific biometrics (e.g., fingerprint) to make the readme more future-proof.
- Added subheadings, restructured, and in some cases rewrote to improve
  clarity.

As part of the changes, fixes the fact that `error_codes.dart` was incorrectly
made private in 2.0, since that change broke one of the examples (in the same
way as it broke plugin clients; `code-excerpt` will prevent something like
that from happening again).

Fixes https://github.com/flutter/flutter/issues/102164
Fixes https://github.com/flutter/flutter/issues/103125

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/local_auth/local_auth/README.md [hunk 1]
```diff
@@ -1,148 +1,188 @@
...
 - BiometricType.face
 - BiometricType.fingerprint
+- BiometricType.weak
+- BiometricType.strong
+
+### Enrolled Biometrics
+
+`canCheckBiometrics` only indicates whether hardware support is available, not
+whether the device has any biometrics enrolled. To get a list of enrolled
+biometrics, call `getAvailableBiometrics()`.
```
Comment (hellohuanlin) [inline packages/local_auth/local_auth/README.md:45]: `canCheckBiometrics` and `getAvailableBiometrics` sounds a bit unclear (that may be the reason we add so many comments to clarify it?)
Maybe `hasAnyBiometricsHardware` and `getEnrolledBiometrics`?

File: packages/local_auth/local_auth/lib/error_codes.dart [hunk 2]
```diff
@@ -9,18 +9,18 @@
 /// PIN/pattern/password (Android) on the device.
 const String passcodeNotSet = 'PasscodeNotSet';
 
-/// Indicates the user has not enrolled any fingerprints on the device.
+/// Indicates the user has not enrolled any biometrics on the device.
 const String notEnrolled = 'NotEnrolled';
 
-/// Indicates the device does not have a Touch ID/fingerprint scanner.
+/// Indicates the device does not have hardware support for biometrics.
```
Comment (hellohuanlin) [inline packages/local_auth/local_auth/lib/error_codes.dart:15]: But previously we use `getAvailableBiometrics` to refer to enrolled biometrics. Maybe "NoBiometricHardware"?

**Review by hellohuanlin — COMMENTED** [review]
Have some concern around API naming. As someone completely new to this plugin I feel that I may have to keep referring back to the doc when writing code. 

Tho this is probably not the scope of this PR.

**Comment by stuartmorgan** [comment]
> Have some concern around API naming. [...] Tho this is probably not the scope of this PR.

Changing the API is definitely not in scope for a PR fixing the README, no.

I agree the names are really confusing in this plugin. In retrospect we probably should have revisited everything since we were doing a breaking change anyway in 2.0, but the focus was on the changes for federation.

If you're interested in doing a 2.1 with a new set of API names for most of these checks, deprecating the old ones (like I did recently with `url_launcher`), I'd be all for that.

**Review by hellohuanlin — APPROVED** [review]

File: packages/local_auth/local_auth/README.md [hunk 3]
```diff
@@ -1,148 +1,188 @@
...
 
+<?code-excerpt path-base="excerpts/packages/local_auth_example"?>
+
 This Flutter plugin provides means to perform local, on-device authentication of
 the user.
 
-This means referring to biometric authentication on iOS (Touch ID or lock code)
-and the fingerprint APIs on Android (introduced in Android 6.0).
+On supported devices, this includes authentication with biometrics such as
+fingerprint of facial recognition.
```
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply packages/local_auth/local_auth/README.md]: Fixed

**Pull request merged**